src/main/java/eu/europeana/metis/sandbox/controller/PatternAnalysisController.java

```diff
@@ -94,13 +103,18 @@ public DatasetProblemPatternAnalysisView<Step> getDatasetPatternAnalysis(
       @ApiResponse(code = 404, message = "Not able to retrieve the pattern analysis of record")
   })
   @GetMapping(value = "{id}/get-record-pattern-analysis", produces = APPLICATION_JSON_VALUE)
-  public List<ProblemPattern> getRecordPatternAnalysis(
+  public ResponseEntity<List<ProblemPattern>> getRecordPatternAnalysis(
       @ApiParam(value = "id of the dataset", required = true) @PathVariable("id") String datasetId,
       @ApiParam(value = "The record content as a file", required = true) @RequestParam String recordId)
       throws SerializationException {
-    String recordContent = recordLogService.getRecordLogEntity(recordId, datasetId, Step.VALIDATE_INTERNAL).getContent();
-    return patternAnalysisService.getRecordPatternAnalysis(rdfConversionUtils.convertStringToRdf(recordContent));
-
+    RecordLogEntity recordLog = recordLogService.getRecordLogEntity(recordId, datasetId, Step.VALIDATE_INTERNAL);
+    if (recordLog != null) {
```

Comment:
I find these kind of simple `if...else` blocks to be unnecessary. I prefer to use the ternary since it's faster 
```suggestion
    return recordLog != null ? new ResponseEntity<>(patternAnalysisService.getRecordPatternAnalysis(rdfConversionUtils.convertStringToRdf(recordLog.getContent())), HttpStatus.OK) : new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Reply:
Let me have a look, is in order to reply the http status when the recordLog is not found, otherwise it throws an unhandled exception.

Reply:
rewrote it in positive so is easy to read and added comments.